external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -243,3 +244,58 @@ def generator():
             yield record
 
     return generator
+
+
+@register_tfdataset_loader
+@dataclasses.dataclass
+class ReservoirTimeSeriesLoader(TFDatasetLoader):
```

Comment:
Please add a docstring.

Could you name this class according to what it does, without mentioning "Reservoir"? While it's custom made to fit your requirements for reservoir training, it would help to make it clear at the level of implementing the data loading what those requirements are. It's also possible this loader will be re-used for other training functions with similar requirements.

To me this seems like a simple NetCDF directory loader with a small feature of being able to enforce dimension ordering different from what's in the files being loaded, and the code is mostly duplicate against NCDirLoader. Is there a reason you didn't add an `Optional[Sequence[str]]` argument to `NCDirLoader` and use that?

Reply:
Moved functionality into NCDirLoader